discovery/src/main/java/com/ibm/watson/developer_cloud/discovery/v1/Discovery.java

```diff
@@ -10,445 +10,721 @@
+    builder.query(VERSION, versionDate);
+    final JsonObject contentJson = new JsonObject();
+    if (updateEnvironmentOptions.name() != null) {
+      contentJson.addProperty("name", updateEnvironmentOptions.name());
+    }
+    if (updateEnvironmentOptions.description() != null) {
+      contentJson.addProperty("description", updateEnvironmentOptions.description());
+    }
+    builder.bodyJson(contentJson);
+
+    return createServiceCall(builder.build(), ResponseConverterUtils.getObject(Environment.class));
+  }
+
+  /**
+   * Query documents.
+   *
+   * See the [Discovery service documentation](https://www.ibm.com/watson/developercloud/doc/discovery/using.html)
+   * for more details.
+   *
+   * @param queryOptions the {@link QueryOptions} containing the options for the call
+   * @return the {@link QueryResponse} with the response
+   */
+  public ServiceCall<QueryResponse> query(QueryOptions queryOptions) {
+    Validator.notNull(queryOptions, "queryOptions cannot be null");
+    RequestBuilder builder = RequestBuilder.get(String.format("/v1/environments/%s/collections/%s/query",
+        queryOptions.environmentId(), queryOptions.collectionId()));
+    builder.query(VERSION, versionDate);
+    if (queryOptions.filter() != null) {
+      builder.query("filter", queryOptions.filter());
+    }
+    if (queryOptions.query() != null) {
+      builder.query("query", queryOptions.query());
+    }
+    if (queryOptions.naturalLanguageQuery() != null) {
+      builder.query("natural_language_query", queryOptions.naturalLanguageQuery());
+    }
+    if (queryOptions.passages() != null) {
+      builder.query("passages", String.valueOf(queryOptions.passages()));
+    }
+    if (queryOptions.aggregation() != null) {
+      builder.query("aggregation", queryOptions.aggregation());
+    }
+    if (queryOptions.count() != null) {
+      builder.query("count", String.valueOf(queryOptions.count()));
+    }
+    if (queryOptions.returnFields() != null) {
+      builder.query("return", queryOptions.returnFields());
+    }
+    if (queryOptions.offset() != null) {
+      builder.query("offset", String.valueOf(queryOptions.offset()));
+    }
+    if (queryOptions.sort() != null) {
+      builder.query("sort", queryOptions.sort());
+    }
+
+    return createServiceCall(builder.build(), ResponseConverterUtils.getObject(QueryResponse.class));
+  }
+
+  /**
+   * Queries for notices (errors or warnings) that may have been generated by the system. Currently, notices are only
+   * generated when ingesting documents. See the [Discovery service documentation](https://www.ibm
+   * .com/watson/developercloud/doc/discovery/using.html) for more details on the query language.
+   *
+   * @param queryNoticesOptions the {@link QueryNoticesOptions} containing the options for the call
+   * @return the {@link NoticeQueryResponse} with the response
+   */
+  public ServiceCall<NoticeQueryResponse> queryNotices(QueryNoticesOptions queryNoticesOptions) {
```

Comment:
I agree that this seems a little odd -- but it's part of the swagger so I'm assuming it's there for a reason.